### PR TITLE
Spring Boot 3.0.x

### PR DESCRIPTION
**Pull request by mayur-solace**
- Upgrade to Spring Boot version from 2.x to 3.x 
- Upgrade the maven plugins to latest versions
- Changed the default value of the reconnectRetries property in solace-java-spring-boot-starter

### CODE REVIEW

File: solace-spring-boot-autoconfigure/solace-java-spring-boot-autoconfigure/pom.xml [hunk 1]
```diff
@@ -25,12 +25,6 @@
 		<dependency>
 			<groupId>org.springframework.boot</groupId>
 			<artifactId>spring-boot-configuration-processor</artifactId>
-			<optional>true</optional>
```
Comment (Nephery) [inline solace-spring-boot-autoconfigure/solace-java-spring-boot-autoconfigure/pom.xml:28]: `spring-boot-configuration-processor` needs to stay optional as per Spring docs: https://docs.spring.io/spring-boot/docs/current/reference/html/configuration-metadata.html#appendix.configuration-metadata.annotation-processor.configuring

File: solace-spring-boot-autoconfigure/solace-jms-spring-boot-autoconfigure/pom.xml [hunk 2]
```diff
@@ -29,17 +29,10 @@
...
-		</dependency>
-		<dependency>
-		    <groupId>org.springframework.boot</groupId>
-			<artifactId>spring-boot-starter-test</artifactId>
-			<scope>test</scope>
 		</dependency>
 		<dependency>
 			<groupId>org.springframework.boot</groupId>
 			<artifactId>spring-boot-configuration-processor</artifactId>
-			<optional>true</optional>
```
Comment (Nephery) [inline solace-spring-boot-autoconfigure/solace-jms-spring-boot-autoconfigure/pom.xml:42]: `spring-boot-configuration-processor` needs to stay optional as per Spring docs: https://docs.spring.io/spring-boot/docs/current/reference/html/configuration-metadata.html#appendix.configuration-metadata.annotation-processor.configuring

File: solace-spring-boot-parent/pom.xml [hunk 3]
```diff
@@ -21,15 +21,17 @@
...
 
-        <solace.spring.boot.java-autoconf.version>4.3.1-SNAPSHOT</solace.spring.boot.java-autoconf.version>
-        <solace.spring.boot.jms-autoconf.version>4.3.1-SNAPSHOT</solace.spring.boot.jms-autoconf.version>
+        <solace.spring.boot.java-autoconf.version>5.0.0-SNAPSHOT</solace.spring.boot.java-autoconf.version>
+        <solace.spring.boot.jms-autoconf.version>5.0.0-SNAPSHOT</solace.spring.boot.jms-autoconf.version>
 
-        <solace.jcsmp.version>10.17.0</solace.jcsmp.version>
-        <solace.jms.version>10.17.0</solace.jms.version>
+        <solace.jcsmp.version>10.19.1-jakarta_jms.43138</solace.jcsmp.version>
+        <solace.jms-jakarta.version>10.19.1-jakarta_jms.43138</solace.jms-jakarta.version>
```
Comment (Nephery) [inline solace-spring-boot-parent/pom.xml:31]: These don't seem to be official release versions: https://mvnrepository.com/artifact/com.solacesystems/sol-jcsmp

File: solace-spring-boot-samples/solace-java-sample-app/pom.xml [hunk 4]
```diff
@@ -19,9 +19,9 @@
 	<properties>
 		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
 		<start-class>demo.DemoApplication</start-class>
-		<java.version>1.8</java.version>
+		<java.version>17</java.version>
 		<spring.boot.version>2.7.7</spring.boot.version>
```
Comment (Nephery) [inline solace-spring-boot-samples/solace-java-sample-app/pom.xml:23]: Needs to be updated to latest `3.x`.

**Review by Nephery — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -22,6 +22,13 @@ For Solace projects on Spring Cloud, please visit the [Solace Spring Cloud](//gi
 
 ## Repository Contents
 
+> Note:
+> 
+> The [master](//github.com/SolaceProducts/solace-spring-boot/tree/master) branch dependencies have been updated to the **Spring Boot 3.x** which requires **JDK 17** or higher.
+>
+> The [master-spring-boot-2.x](//github.com/SolaceProducts/solace-spring-boot/tree/master-spring-boot-2.x) branch will act as master branch for solace-spring-boot compatible with **Spring Boot 2.x**.
```
[the remark this answers is not in the document]
Comment (mayur-solace) [reply README.md]: Fixed.

File: solace-spring-boot-autoconfigure/solace-jms-spring-boot-autoconfigure/pom.xml (hunk 2 above)
Comment (mayur-solace) [reply solace-spring-boot-autoconfigure/solace-jms-spring-boot-autoconfigure/pom.xml]: Fixed.

File: solace-spring-boot-samples/solace-java-sample-app/pom.xml (hunk 4 above)
Comment (mayur-solace) [reply solace-spring-boot-samples/solace-java-sample-app/pom.xml]: Fixed.

File: solace-spring-boot-autoconfigure/solace-java-spring-boot-autoconfigure/pom.xml (hunk 1 above)
Comment (mayur-solace) [reply solace-spring-boot-autoconfigure/solace-java-spring-boot-autoconfigure/pom.xml:28]: Fixed.

**Review by Nephery — APPROVED** [review]
JCSMP/JMS versions will be updated to release version whenever they come out.

**Pull request merged**